HILCTA/PolinomCalculator/PolinomCalculator/Polinom.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PolinomCalculator
+{
+    public class Polinom
+    {
+        public List<double> Numbers { get; set; }
+
+
+        public Polinom(params double[] args)
+        {
+            Numbers = new List<double>(args);
+            RemoveLeadingZeros();
+        }
+
+        public Polinom(int monomialDegree, bool isMonomial = false)
+        {
+            if (isMonomial)
+            {
+                Numbers = new List<double> { 1 };
+                for (int i = 0; i < monomialDegree; i++)
+                {
+                    Numbers.Add(0);
+                }
+            }
+            else
+            {
+                Numbers = new List<double> { monomialDegree };
+            }
+
+        }
+
+
+        public Polinom Clone()
+        {
+            return new Polinom(Numbers.ToArray());
+        }
+
+
+        public Polinom Derivative()
+        {
+            var result = Clone();
+            if (result.Numbers.Count > 0)
+            {
+                result.Numbers.RemoveAt(result.Numbers.Count - 1);
+            }
+
+            int multiplier = result.Numbers.Count;
+            for (int i = 0; i < result.Numbers.Count; i++)
+            {
+                result.Numbers[i] *= multiplier;
+                multiplier--;
+            }
+
+            return result;
+        }
+
+        private void RemoveLeadingZeros()
+        {
+            while (Numbers.Count > 0 && Numbers[0] == 0)
+            {
+                Numbers.RemoveAt(0);
+            }
+        }
+
+        public double ValueAt(double x)
+        {
+            double result = 0;
+            for (int i = 0; i < Numbers.Count; i++)
+            {
+                result += Numbers[i] * Math.Pow(x, Numbers.Count - 1 - i);
+            }
+            return result;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Polinom other)
+            {
+                RemoveLeadingZeros();
```

Comment:
Öö :D Az Equals módosítja az objektum belső állapotát a RemoveLeadingZeros() hívással?? Ennek nem így kellene működnie. Ha a RemoveLeadingZeros() szükséges, akkor azt miért nem a létrehozáskor hívod? Az Equals hívásnak NEM szabad módosítania belső állapotot. Ez nagyon durva Single responsibility és Liskov elv sértés is.